LaTeX/LaTeX.sublime-syntax

```diff
@@ -110,79 +110,89 @@ contexts:
-        # start of actual command definition
-        (\{)
+    - match: '(\\)(?:(?:new|renew|provide)command\*?)'
+      scope: meta.function.latex keyword.declaration.function.latex storage.modifier.newcommand.latex
       captures:
-        1: support.function.newcommand.latex storage.modifier.newcommand.latex
-        2: punctuation.definition.backslash.latex
-        3: punctuation.definition.group.brace.begin.latex
-        4: support.function.latex entity.name.newcommand.latex
-        5: punctuation.definition.group.brace.end.latex
-        6: support.function.latex entity.name.newcommand.latex
-        7: punctuation.definition.group.bracket.begin.latex
-        8: meta.number.integer.decimal.latex constant.numeric.value.latex
-        9: punctuation.definition.group.bracket.end.latex
-        10: punctuation.definition.group.bracket.begin.latex
-        11: meta.function.parameters.default-value.latex
-        12: punctuation.definition.group.bracket.end.latex
-        13: punctuation.definition.group.brace.begin.latex
+        1: punctuation.definition.backslash.latex
       push:
-        - meta_scope: meta.function.newcommand.latex
-        - match: '\}'
-          scope: punctuation.definition.group.brace.end.latex
+        - newcommand-definition
+        - newcommand-argspec
+        - newcommand-commandname
+    - match: '(\\)DeclareMathOperator\*?'
+      scope: meta.function.latex support.function.declare-math-operator.latex storage.modifier.newcommand.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - newcommand-definition
+        - newcommand-commandname
+
+  newcommand-argspec:
+    - match: '(\[)\s*(\d)\s*(\])'
+      scope: meta.function.parameters.latex
+      captures:
+        1: punctuation.definition.group.bracket.begin.latex
+        2: constant.numeric.value.latex meta.number.integer.decimal.latex
+        3: punctuation.definition.group.bracket.end.latex
+      push:
+        - match: '\['
+          scope: punctuation.definition.group.bracket.begin.latex
+          push:
+            - meta_scope: meta.function.parameters.default-value.latex
+            - match: '\]'
+              scope: punctuation.definition.group.bracket.end.latex
+              pop: 2
+            - include: general-constants
+            - include: general-commands
+            - include: macro-braces
+        - match: '(?=\S)'
           pop: true
-        - include: general-constants
-        - include: general-commands
-        - include: macro-braces
-    - match: |-
-        (?x)
-        (
-          (\\)
-          DeclareMathOperator\*?
-        )
-        \s*
-        (?:
-          (\{)\s*(\\[A-Za-z@]+)\s*(\})
-          | (\\[A-Za-z@]+)
-        )
-        \s*
-        (\{)
+
+    # if we enounter anything that is not an opening bracket, whitespace, or
+    # a comment, then there are no argument specifications and we can continue
+    # with the definition
+    - match: '(?=\S)'
```

Comment:
I suggest pulling this out into a utility include, if there isn't one already. I think this one has a particular name in other syntax files, maybe `immediately-pop`? Then include where relevant.